src/coguard_cli/discovery/config_file_finders/config_file_finder_kubernetes.py

```diff
@@ -0,0 +1,160 @@
+"""
+This module contains the class to find Kubernetes configurations
+inside a folder structure.
+"""
+
+import os
+import re
+import shutil
+import tempfile
+import logging
+from typing import Dict, List, Optional, Tuple
+import yaml
+from flatten_dict import unflatten
+from coguard_cli.discovery.config_file_finder_abc import ConfigFileFinder
+import coguard_cli.discovery.config_file_finders as cff_util
+from coguard_cli.print_colors import COLOR_CYAN, COLOR_TERMINATION
+
+class ConfigFileFinderKubernetes(ConfigFileFinder):
+    """
+    The class to find kubernetes configuration files within a file system.
+    """
+
+    def _create_temp_location_and_mainfest_entry(
+            self,
+            path_to_file_system: str,
+            file_name: str,
+            location_on_current_machine: str) -> Tuple[Dict, str]:
+        """
+        Common helper function which creates a temporary folder location for the
+        configuration files, and then analyzes include directives. It returns
+        a tuple containing a manifest for a kubernetes service and the path to the
+        temporary location.
+        """
+        temp_location = tempfile.mkdtemp(prefix="coguard-cli-kubernetes")
+        to_copy = cff_util.get_path_behind_symlinks(
+            path_to_file_system,
+            location_on_current_machine
+        )
+        shutil.copy(
+            to_copy,
+            os.path.join(
+                temp_location,
+                os.path.basename(location_on_current_machine)
+            )
+        )
+        manifest_entry = {
+            "version": "1.0",
+            "serviceName": "kubernetes",
+            "configFileList": [
+                {
+                    "fileName": file_name,
+                    "defaultFileName": "kube-deployment.yaml",
+                    "subPath": ".",
+                    "configFileType": "yaml"
+                }
+            ],
+            "complimentaryFileList": []
+        }
+        return (
+            manifest_entry,
+            temp_location
+        )
```

Comment:
maybe it is time to move this one out and create a utility function out of it for this specific type of config file.

Reply:
Done